### DISCUSSION TITLE
Trying to run sample script on Raspberry Pi 3

### DISCUSSION THREAD

**Discussion by Bravo13** [pyopencl/pyopencl#714] [Troubleshooting]
SO, I'm trying to run example from page https://documen.tician.de/pyopencl/index.html on RPi 3 with VC4C. Mine clinfo shows I have video core. Also, simple C program works ok. But with python mine program stuck. If I'm running exactly as on example, it hangs on `cl.enqueue_copy(queue, res_np, res_g)` line. If I'm adding `.wait()` to `knl(queue, a_np.shape, None, a_g, b_g, res_g)` it hangs there. Need advice on what possible solution coudl be here or is it possible to look for anu debug data

**Comment by inducer** [comment]
Some things to try:

- set the environment variable `PYOPENCL_NO_CACHE=1`.
- Build PyOpenCL with tracing: In the source directory, do `./configure.py --cl-trace`, `rm -Rf build`, `pip install .`. This will show you which CL call hangs.

**Comment by Bravo13** [comment]
Hey @inducer thanks for your fast reply. Result:
* adding `PYOPENCL_NO_CACHE=1` did not change anything
* building with `--cl-trace` produce next logclGetPlatformIDs
clGetPlatformIDs
clGetPlatformInfo
clGetPlatformInfo
Choose platform:
[0] <pyopencl.Platform 'OpenCL for the Raspberry Pi VideoCore IV GPU' at 0x129518c>
Choice [0]:clGetDeviceIDs
clGetDeviceIDs
clCreateContext
clGetContextInfo
clGetContextInfo
clGetContextInfo
clGetContextInfo
clGetDeviceInfo
clGetPlatformInfo
clGetPlatformInfo
clCreateCommandQueue
clCreateBuffer
gpuserv: vc_gpuserv_init: starting initialisation
clCreateBuffer
clGetContextInfo
clGetContextInfo
clGetDeviceInfo
clGetPlatformInfo
clGetPlatformInfo
clCreateProgramWithSource
clBuildProgram
clGetProgramInfo
clGetProgramInfo
clGetProgramBuildInfo
clGetProgramBuildInfo
clCreateBuffer
clCreateKernel
clGetKernelInfo
clGetKernelInfo
clGetKernelInfo
clGetKernelInfo
clGetKernelInfo
clRetainContext
clGetContextInfo
clGetContextInfo
clReleaseContext
clGetDeviceInfo
clGetDeviceInfo
clGetDeviceInfo
clGetPlatformInfo
clGetPlatformInfo
clGetKernelInfo
clSetKernelArg
clSetKernelArg
clSetKernelArg
clEnqueueNDRangeKernel
clReleaseEvent
clGetMemObjectInfo
clEnqueueReadBuffer

**Comment by Bravo13** [comment]
Just found there is some additional debug from VC4CL with `VC4CL_DEBUG=events,execution PYOPENCL_NO_CACHE=1 python test.py` So, I attached new log(too big to put in message, I suppose). Have a feeling it's not a problem of PyOpenCl itself, but some conditions that makes vc4cl fail.
[log.txt](https://github.com/inducer/pyopencl/files/14177589/log.txt)
[test.py.txt](https://github.com/inducer/pyopencl/files/14177608/test.py.txt)

**Reply by Bravo13** [reply]
And after 5000000000 us log looks like this(see attachements)
[log.txt](https://github.com/inducer/pyopencl/files/14181096/log.txt)
